### DISCUSSION TITLE
Configurazione automatica del certificato -  Let's Encrypt

### DISCUSSION THREAD

**Discussion by LorenzoPetraroli** [globaleaks/globaleaks-whistleblowing-software#3608] [Q&A]
### What version of GlobaLeaks are you using?

4.12.8

### What browser(s) are you seeing the problem on?

Chrome, Firefox

### What operating system(s) are you seeing the problem on?

Linux

### Describe the issue

Abbiamo provato ad avviare la Configurazione Automatica per generare il certificato https facendo uso della Certification Authority Let's Encrypt. 
La configurazione rimane bloccata in quanto viene generato l'errore (riscontrato nei log) riportato di seguito.

---------------------------------

Platform: 
Host: whistleblowing.xxxxx.cloud (a5mdcz3l2smxumbwqyu3kblnuk3k5isikd2uxnu2xluirtl3j7ngmyyd.onion)
Version: 4.12.8

AttributeError Attribute not found.

Traceback (most recent call last):

  File "/root/GlobaLeaks/backend/env/lib/python3.10/site-packages/twisted/internet/defer.py", line 1418, in _inlineCallbacks
    result = g.send(result)

  File "/root/GlobaLeaks/backend/globaleaks/rest/api.py", line 425, in concludeHandlerFailure
    self.handle_exception(err, request)

  File "/root/GlobaLeaks/backend/globaleaks/rest/api.py", line 252, in handle_exception
    e.tid = request.tid

  File "/root/GlobaLeaks/backend/env/lib/python3.10/site-packages/josepy/util.py", line 191, in _setattr_
    raise AttributeError("can't set attribute")

AttributeError: can't set attribute

-------------------------------------

Potete darci qualche informazione a riguardo?
Grazie

**Comment by evilaliv3** [comment]
Grazie @LorenzoPetraroli , penso che l'eccezione che segnali sia di fatto una eccezione nell'eccezione che abbiamo risolto e stiamo per pubblicar.e

In realtà però credo che il vostro problema sia causato da altro:
- avete puntato il DNS verso il vostro server?
- avete un ip pubblico su cui globaleaks è esposto sia su porta 80 che su porta 443?

**Reply by LorenzoPetraroli** [reply]
Grazie per la risposta rapida, si sono stati seguiti entrambi gli step, aspettiamo l'aggiornamento allora

**Comment by evilaliv3** [comment]
@LorenzoPetraroli : l'aggiornamento vi farebbe solo vedere la vera eccezione, ma non vi risolverà il problema.

Se puoi fornirmi il dominio provvedo a verificare dal mio punto di vista se riesco aidentificare il problema.

**Reply by LorenzoPetraroli** [reply]
whistleblowing.parsec326.cloud

**Comment by evilaliv3** [comment]
@LorenzoPetraroli : port 80 and 443 seems actually close now. is this intentional?

**Reply by gianluca326** [reply]
Buongiorno, abbiamo risolto installando ngnix, ora funziona correttamente.
Grazie del supporto.
Saluti

**Comment by evilaliv3** [comment]
Grazie dell'aggiornamento.

Vi informiamo che l'uso di Nginx o di altri eventuali proxy di fronte
all'applicativo invalidano le misure di sicurezza implementate
dall'applicativo come tale vi suggeriamo di rimuoverlo ed esporre
l'applicazione direttamente

**Reply by gianluca326** [reply]
Grazie mille, l'uso è temporaneo, vorrei sapere dove disponibile la documentazione delle API.

**Reply by evilaliv3** [reply]
No al momento non c'è una documentazione pubblica delle API in quantoi seppur l'applicativo sia interamente ralizzato con una API REST questa viene usata dai componenti ufficiali del progetto.

Stiamo prevedendo documentazione ed estensione per prevedere che il sofware sia collegabile ad altri software ma ancora non abbiamo nulla di pubblic. Se poteste chiarirci il vostro requisito ciò ci sarebbe molto utile grazie

**Reply by gianluca326** [reply]
Con la vecchia versione di globaleaks avevamo a disposizione delle API che ci hanno permesso di interfacciarlo col ns gestionale, per la lettura delle segnalazioni, quindi credevo fossero a disposizione anche per l'attuale versione.

**Comment by evilaliv3** [comment]
Quello che c'era c'è certamernente. Probabilmente vi eravate fatti un po di reversing delle API usate dal software che sono tutte API REST più o meno standard

**Reply by gianluca326** [reply]
Prima utilizzavo il metodo 

AUTENTICAZIONE (Ricevente)
URL richiesta →
https://whistleblowing.parsec326.it/authentication
Metodo di richiesta →
POST
BODY →
{"username":"recipient","password":"recipient","token":""}
RISPOSTA →
{
"status": "enabled",
"user_id": "5105ae95-ec66-4c34-b050-922c1ba1ca72",
"password_change_needed": false,
"session_id": "KuqXSTkX5xDwqlLpOSEGaHsfRmPHBp8cfhf6rEQpnR",
"role": "receiver",
"session_expiration": 1523873690
}

se lo richiamo dalla nuova versione mi da:

{
   "error_message": "Method not implemented",
   "error_code": 2,
   "arguments": []
}

**Comment by evilaliv3** [comment]
L'API esiste ancora ma alcune hanno cambiato nome nell'obbiettivo appunto
di creare una API più chiara e utilizzabile pubblicamente.

Chi vi ha supportato nella presente implementazione potrebbe aggiornervela
riprendendo mano alla corrente implementazione. Le API sono definite in
codice qui:
https://github.com/globaleaks/GlobaLeaks/blob/main/backend/globaleaks/rest/api.py

**Reply by gianluca326** [reply]
Grazie, nel frattempo avevo individuato questo file all'interno del progetto, sembra che ci sia tutto, vorrei solo sapere se i parametri per la request sono rimasti invariati per i vari metodi, o se sono variati (o aggiunti/rimossi).
C'è un modo per risalire ai parametri?
Grazie mille per la disponibilità.

**Comment by evilaliv3** [comment]
Grazie per la domanda @LorenzoPetraroli 

Se volete avviare una discussione su questo argomento eventualmente potreste aviarla come discussione indipendente e specifica? Chiudo questa come gestita.
